skfp/descriptors/topological.py

```diff
@@ -180,6 +181,47 @@ def graph_distance_index(mol: Mol, distance_matrix: Optional[np.ndarray] = None)
     return int(sum((k * f) ** 2 for k, f in distance_counts.items()))
 
 
+def hall_kier_alpha(mol: Mol) -> int:
+    r"""
+    Hall-Kier molecular flexibility index (alpha).
+
+    The Hall-Kier alpha index [1]_ is a molecular descriptor that accounts for molecular
+    flexibility based on the number of heavy atoms and the number of bonds in a molecule.
+    It is calculated as:
+
+    .. math::
+        \alpha = A - B + 1
```

Comment:
This is wrong descriptor and description. See RDKit code: https://github.com/rdkit/rdkit/blob/44364fd9824b37f47e62ce53d6f8d32b020d7da9/rdkit/Chem/GraphDescriptors.py#L76.

Kier flexibility index `phi` is defined as `kappa1_alpha * kappa2_alpha / n_atoms`, source publication is https://onlinelibrary.wiley.com/doi/abs/10.1002/qsar.19890080307

Hall-Kier alpha is in RDKit, function `HallKierAlpha`, and returns the value of `alpha` from the paper above:
```
The value of alpha is taken from the ratio of covalent radii of atom x relative to Csp^3, thus:
a = r/r(csp3) - 1
```